scio-core/src/main/scala/com/spotify/scio/ScioResult.scala

```diff
@@ -72,10 +73,20 @@ abstract class ScioResult private[scio] (val internal: PipelineResult) {
 
   /**
    * Wait until the pipeline finishes with the State `DONE` (as opposed to `CANCELLED` or
-   * `FAILED`). Throw exception otherwise.
+   * `FAILED`). Throw exception otherwise. If `doCancelJob` is set, cancel the [[PipelineResult]].
    */
-  def waitUntilDone(duration: Duration = Duration.Inf): ScioResult = {
-    waitUntilFinish(duration)
+  def waitUntilDone(duration: Duration = Duration.Inf, doCancelJob: Boolean = false): ScioResult = {
+    try {
```

Comment:
what about moving this to `waitUntilFinish`, since they both expect a duration and `waitUntilDone` relies on it. wdyt?

Reply:
I can do that! I wasn't 100% sure of the semantic implications of each method tbh